README.md

```diff
@@ -53,54 +121,106 @@ The source for these tags is at [_plugins/links.rb](_plugins/links.rb).
 | ghc   | GitHub code link          | Branch defaults to `gh_branch` setting in `_config.yml`. Override using `-b` | `{% ghc server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java %}` `{% ghc -b 1.9 README.md %}` |
 | jira   | Jira issue link          | None  | `{% jira ACCUMULO-1000 %}` |
 
-## Updating property documentation
 
-Building Accumulo  generates `server-properties.md` and `client-properties.md`.  To
-regenerate these, do the following.
+## Updating Property Documentation
 
-```
-cd <accumulo source dir>
+Building Accumulo  generates `server-properties.md` and `client-properties.md`.
+To regenerate these, do the following.
+
+```bash
+ACCUMULO_SITE_CLONE=<accumulo website clone location, with master branch checked out>
+ACCUMULO_CLONE=<accumulo clone location>
+cd "$ACCUMULO_CLONE"
 mvn package -DskipTests
-cp ./core/target/generated-docs/server-properties.md <accumulo website source>/_docs-2/configuration
-cp ./core/target/generated-docs/client-properties.md <accumulo website source>/_docs-2/configuration
+cp ./core/target/generated-docs/server-properties.md "$ACCUMULO_SITE_CLONE"/_docs-2/configuration
+cp ./core/target/generated-docs/client-properties.md "$ACCUMULO_SITE_CLONE"/_docs-2/configuration
 ```
 
-## Update the production website
+## Staging the Generated Site
 
-For Apache Accumulo committers, the `asf-site` branch needs to be updated with the generated
-HTML.  Changes to this branch are automagically mirrored to the website.
+For Apache Accumulo committers, the `asf-staging` branch needs to be updated
+with the generated HTML. Changes to this branch are automagically mirrored to
+the [staging website][staging].
 
-This can be done easily by invoking the post-commit hook (either by hand, or automatically via configuring
-Git to invoke the post-commit hook).  The commands below are a guide for committers who wish to publish
-the web site.
+This can be done easily by invoking the post-commit hook (either by hand, or
+automatically via configuring Git to invoke the post-commit hook).  The
+commands below are a guide for committers who wish to stage the web site.
 
 ```bash
-# ensure local asf-site branch is up to date
-git checkout asf-site
-git pull upstream asf-site
-
-# switch to master branch, update it, and build new site
+# Step 0: stay in master branch; you never need to switch
 git checkout master
-git pull upstream master
+
+# Step 1: update your upstream remote
+git remote update upstream
+
+# Step 2: ensure your local master branch is up-to-date
+# Note: this will destroy any local changes; create a branch first if you want to keep any of that
+git reset --hard upstream/master
+
+# Step 3: create/update your local asf-staging branch
+git branch --track --force asf-staging upstream/asf-staging
```

Comment:
What is the benefit of using asf-staging?  This seems to complicate the process of updating the website, especially for small changes.  I guess this would be useful for larger changes like re-organizing pages, changing layout, etc.

Reply:
The main benefit for now is that you can test the change before you publish it. In future, this benefit may be more obvious when the automated Jekyll building-to-staged stuff is working, at which point, most of this workflow goes away and becomes a simple 3 step process:

1. merge PR in GitHub UI,
2. view automatically staged site to confirm the change (https://accumulo.staged.apache.org),
3. execute simple git command to update asf-site from asf-staging: `git remote update && git push upstream refs/remotes/upstream/asf-staging:asf-site`

Reply:
> In future, this benefit may be more obvious when the automated Jekyll building-to-staged stuff is working, at which point, most of this workflow goes away and becomes a simple 3 step process

Perhaps we should just wait until then to make changes to the workflow.

Reply:
I can modify this to use asf-site again instead of asf-staging for now.

Reply:
Rather than modify this to use asf-site, I'm going to cut this entire section out, since I'm working with INFRA on https://issues.apache.org/jira/browse/INFRA-19737 to try to get us set up to automatically build the site when we merge to master. When fully working, the only thing we'll need to do in git is copy the commits in asf-staging to asf-site when we're happy the site is staged correctly.